server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilderTests.java

```diff
@@ -36,10 +36,11 @@ private DateHistogramValuesSourceBuilder randomDateHistogramSourceBuilder() {
             histo.script(new Script(randomAlphaOfLengthBetween(10, 20)));
         }
         if (randomBoolean()) {
-            histo.dateHistogramInterval(randomFrom(DateHistogramInterval.days(10),
+            histo.calendarInterval(randomFrom(DateHistogramInterval.days(1),
```

Comment:
Maybe we should test the legacy interval here too?

Reply:
I'm not sure how to get the warnings to play nice.  E.g. if this creates a legacy interval, it will throw a deprecation warning in the tests executed by `BaseAggregationTestCase`, so there's no opportunity to `assertWarnings()`.

I could store what kind of interval was chosen in a static variable and then enable/disable warning checks based on that.  A bit better than nothing?

Reply:
Enabling and disabling warnings based on a static variable seems fragile to me. Maybe we should create a new class `LegacyCompositeAggregationBuilderTests` which will build histos with legacy intervals and then we can override the test methods from `BaseAggregationTestCase`, call the super method and then assert the warning. WDYT?

Reply:
I like that approach.  Seems much saner than my suggestion :)